Templates/cmake/README.md

```diff
@@ -0,0 +1,125 @@
+
+>**Note**: CMake support for GNU style OneAPI driver `icpx` on Windows is not yet available until CMake 3.25 and such support will be added in future releases of CMake.
+>**Note**: For comprehensive information about oneAPI programming, see the *[Intel® oneAPI Programming Guide](https://software.intel.com/en-us/oneapi-programming-guide)*. (Use search or the table of contents to find relevant information quickly.)
+>**Note**: For comprehensive information about CMake with OneAPI Compilers, see the *[Intel® oneAPI DPC++/C++ Compiler Developer Guide and Reference](https://www.intel.com/content/www/us/en/docs/dpcpp-cpp-compiler/developer-guide-reference)*. (Use search or the table of contents to find relevant information quickly. OR, Navigate to Compiler Setup -> Use the Command Line -> Use the Cmake with the compiler)
+
+
+## Set Environment Variables
+
+When working with the command-line interface (CLI), you should configure the oneAPI toolkits using environment variables. Set up your CLI environment by sourcing the `setvars` script every time you open a new terminal window. This practice ensures that your compiler, libraries, and tools are ready for development.
+
+Ensure the availability of latest CMake binaries.
+
+## Build the Cmake example Sample
+
+> **Note**: If you have not already done so, set up your CLI
+> environment by sourcing  the `setvars` script in the root of your oneAPI installation.
+>
+> Linux*:
+> - For system wide installations: `. /opt/intel/oneapi/setvars.sh`
+> - For private installations: ` . ~/intel/oneapi/setvars.sh`
+> - For non-POSIX shells, like csh, use the following command: `bash -c 'source <install-dir>/setvars.sh ; exec csh'`
+>
+> Windows*:
+> - `C:\Program Files (x86)\Intel\oneAPI\setvars.bat`
+> - Windows PowerShell*, use the following command: `cmd.exe "/K" '"C:\Program Files (x86)\Intel\oneAPI\setvars.bat" && powershell'`
+>
+> For more information on configuring environment variables, see *[Use the setvars Script with Linux* or macOS*](https://www.intel.com/content/www/us/en/develop/documentation/oneapi-programming-guide/top/oneapi-development-environment-setup/use-the-setvars-script-with-linux-or-macos.html)* or *[Use the setvars Script with Windows*](https://www.intel.com/content/www/us/en/develop/documentation/oneapi-programming-guide/top/oneapi-development-environment-setup/use-the-setvars-script-with-windows.html)*.
+
+### On Linux*
+
+1. Change to the sample directory.
+2. Build the program.
+   ```
+    $ mkdir build
+    $ cd build
+    $ CC=icx CXX=icpx FC=ifx cmake ..
+    $ cmake --build . -j
+   ```
+
+### On Windows*
+
+
+1. Change to the sample directory.
+2. Build the program.
+   ```
+    $ mkdir build
+    $ cd build
+    $ CC=icx CXX=icx FC=ifx cmake -GNinja ..
+    $ cmake --build . -j
+   ```
+>**Note**: Currently, only Ninja generators are supported in the Windows. A Visual Studio Generator may be supported with OneAPI may be supported in the future CMake releases.

```

Comment:
```suggestion
>**Note**: Currently, only Ninja generators are supported on Windows.
```

I don't think that is true as all the FPGA samples use "NMake Makefiles" as a generator

Reply:
I meant the support we added/maintained.

Reply:
done